packages/fuel-gauge/src/launch-test-node-types.test.ts

```diff
@@ -0,0 +1,29 @@
+import { launchTestNode } from 'fuels/test-utils';
+
+import type { AdvancedLoggingAbi, CallTestContractAbi } from '../test/typegen';
+import { AdvancedLoggingAbi__factory, CallTestContractAbi__factory } from '../test/typegen';
+import AdvancedLoggingBytecode from '../test/typegen/contracts/AdvancedLoggingAbi.hex';
+import CallTestContractBytecode from '../test/typegen/contracts/CallTestContractAbi.hex';
+
+/**
+ * @group node
+ * @group browser
+ */
+describe('type level tests for launchTestNode', () => {
+  test('infers types correctly', async () => {
+    using launched = await launchTestNode({
+      contractsConfigs: [
+        { deployer: AdvancedLoggingAbi__factory, bytecode: AdvancedLoggingBytecode },
+        { deployer: CallTestContractAbi__factory, bytecode: CallTestContractBytecode },
+        { deployer: AdvancedLoggingAbi__factory, bytecode: AdvancedLoggingBytecode },
+      ],
+    });
+    const {
+      contracts: [c1, c2, c3],
+    } = launched;
+
+    expectTypeOf(c1).toMatchTypeOf<AdvancedLoggingAbi>();
```

Comment:
Apparently these tests pass regardless of the expected type matching the real type or not. I'm currently trying to get them to fail. If @FuelLabs/sdk-ts think that we should merge it for the immediate benefit of the PR and that I investigate it in another PR, I'm down for that as well as holding the PR until the tests actually fail when they should.

Reply:
@nedsalk This is ok for now. I've tested this locally and the types are being properly inferred.

<img width="740" alt="Screenshot 2024-07-11 at 12 26 33" src="https://github.com/FuelLabs/fuels-ts/assets/30977845/5fce2acb-350b-49e0-b87b-eaa83b209f6d">
<img width="904" alt="Screenshot 2024-07-11 at 12 26 50" src="https://github.com/FuelLabs/fuels-ts/assets/30977845/0871ba08-2895-4166-9543-e0f44c26784e">

Reply:
I don't believe there is a way to actually unit test this because Type inferencing is done statically and types don't exist at runtime. I suggest we remove this test because the proof this works is done by the language server

Reply:
@nedsalk I'm getting TS errors if it isn't the correct type - IMO this is sufficient :)

<img width="1311" alt="image" src="https://github.com/FuelLabs/fuels-ts/assets/16990131/36585485-4065-4a16-aea5-3db9df9f3ce9">

Reply:
> Upon reconsideration, I don't believe there is a way to actually unit test this because Type inferencing is done statically and types don't exist at runtime. I suggest we remove this test because the proof this works is done by the language server

`vitest` has type-level testing capabilities, I just need to figure out how to use it. Apparently it's still experimental as well. I know of a different way to have these tests by using `tsc` directly, but I'll first try to use `vitest`.

Reply:
Awesome!